lib/api/src/Cardano/Wallet/Api/Http/Server/Error.hs

```diff
@@ -1021,38 +1045,53 @@ instance IsServerError ErrBalanceTxOutputError where
                     toWalletCoin minimumExpectedCoin
                 }
         ErrBalanceTxOutputSizeExceedsLimit {output} ->
-            apiError err403 OutputTokenBundleSizeExceedsLimit $ mconcat
-                [ "One of the outputs you've specified contains too many "
-                , "assets. Try splitting these assets across two or more "
-                , "outputs. Destination address: "
-                , pretty address
-                , ". Asset count: "
-                , pretty assetCount
-                , "."
-                ]
+            flip (apiError err403) errorMessage $ OutputTokenBundleSizeExceedsLimit
+            ApiErrorOutputTokenBundleSizeExceedsLimit
+                { address = toText address
+                , bundleSize = assetCount
+                }
               where
                 address = toWalletAddress (fst output)
                 assetCount = TokenMap.size $
                     toWalletTokenBundle (snd output) ^. #tokens
+                errorMessage = T.unwords
+                    [ "One of the outputs you've specified contains too many "
+                    , "assets. Try splitting these assets across two or more "
+                    , "outputs. Destination address: "
+                    , pretty address
+                    , ". Asset count: "
+                    , pretty assetCount
+                    , "."
```

Comment:
Similar suggestion: make the error message a constant.

```suggestion
                    , "outputs."
```

Reply:
:+1: